lib/mvi/responses/add_parser.rb

```diff
@@ -56,11 +56,33 @@ def parse
         raw_codes = locate_elements(@original_body, ACKNOWLEDGEMENT_DETAIL_CODE_XPATH)
         return [] unless raw_codes
 
-        raw_codes.map(&:attributes)
+        attributes = raw_codes.map(&:attributes)
+        parse_ids(attributes)
       end
 
       private
 
+      def parse_ids(attributes)
+        codes = { other: [] }
+        attributes.each do |attribute|
+          if attribute[:code].include? 'BRLS'
+            codes[:birls_id] = sanitize_ids(attribute[:code])
+          elsif attribute[:code].include? 'CORP'
+            codes[:participant_id] = sanitize_ids(attribute[:code])
+          else
+            codes[:other].append(attribute)
+          end
```

Comment:
Here's another nitpicky syntax suggestion you can feel free to push back on:
```suggestion
          case attribute[:code]
          when /BRLS/
            codes[:birls_id] = sanitize_ids(attribute[:code])
          when /CORP/
            codes[:participant_id] = sanitize_ids(attribute[:code])
          else
            codes[:other].append(attribute)
          end
```

Reply:
Case statement using regexes, I like it!